iam-role/iam-role.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/acorn-io/services/aws/libs/common"
+	"github.com/aws/aws-cdk-go/awscdk/v2"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awsiam"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/aws/constructs-go/constructs/v10"
+	"github.com/aws/jsii-runtime-go"
+	"github.com/sirupsen/logrus"
+)
+
+type IAMRoleStackProps struct {
+	StackProps                awscdk.StackProps
+	Tags                      map[string]string `json:"tags"`
+	RoleName                  string            `json:"roleName"`
+	TrustedArn                string            `json:"trustedArn"`
+	MaxSessionDurationMinutes int               `json:"maxSessionDurationMinutes"`
+	Path                      string            `json:"path"`
+	ExternalIds               string            `json:"externalIds"`
```

Comment:
you should be able to add Policies here as a map[string]interface{} and pick it up when this is loaded.